akka-persistence-typed/src/main/scala/akka/persistence/typed/internal/RetentionCriteriaImpl.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (C) 2019 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.persistence.typed.internal
+
+import akka.annotation.InternalApi
+import akka.persistence.typed.scaladsl
+import akka.persistence.typed.javadsl
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] final case class SnapshotRetentionCriteriaImpl(
```

Comment:
This doesn't seem evolvable. The original idea was to allow multiple algorithms and inputs. This by name suggests snapshot retention is only based on count (num events and snapshots) versus allowing a TTL or size in MB for example. If this is meant to be just the first of a few impls, I'd get rid of the 'impl' part and rename this to something reflecting that it only handles a count.

Reply:
Ok, can rename to SnapshotCountRetentionCriteria

Other types can be added for the other algorithms in the future.

Reply:
That sounds clearer thanks